packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -453,6 +456,145 @@ export class PrepareProvider extends Provider implements PreparationProvider {
     };
   }
 
+  /**
+   *
+   * @description Method to prepare a stake pseudo-operation
+   * @param operation RPCOperation object or RPCOperation array
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  async stake({ fee, storageLimit, gasLimit, ...rest }: StakeParams): Promise<PreparedOperation> {
+    const { pkh } = await this.getKeys();
+
+    const protocolConstants = await this.context.readProvider.getProtocolConstants('head');
+    const DEFAULT_PARAMS = await this.getOperationLimits(protocolConstants);
+    const op = await createTransferOperation({
+      ...rest,
+      to: pkh,
+      ...mergeLimits({ fee, storageLimit, gasLimit }, DEFAULT_PARAMS),
+      parameter: {
+        entrypoint: 'stake',
+        value: {
+          prim: 'Unit',
+        },
+      },
+    });
+
+    const operation = await this.addRevealOperationIfNeeded(op, pkh);
```

Comment:
theoretically an un-reavled address can't stake as they never set/register a delegate in the first place which is a manager op (should reveal), do we want to throw error in taquito or let octez-client throw their error? (same apply to other two pseudo ops)